addons/loyalty/views/portal_template.xml

```diff
@@ -0,0 +1,97 @@
+                                class="img-fluid"
+                                t-att-src="'/loyalty/static/src/img/award.svg'
+                                    if program == 'loyalty'
+                                    else '/loyalty/static/src/img/wallet.svg'"
+                                width="40"
+                            />
+                            <div class="d-flex flex-column gap-1">
+                                <div>
+                                    <strong t-out="card.program_id.display_name"/>
+                                </div>
+                                <p id="card_points" class="m-0 text-600">
+                                    <t t-out="card.points_display"/>
+                                </p>
+                                <t t-if="any(reward.required_points &lt;= card.points for reward in card.program_id.reward_ids) and program == 'loyalty'">
+                                    <p class="m-0 text-info">A reward is waiting for you</p>
+                                </t>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </template>
+
+    <template id="portal_my_home_loyalty_history" name="Portal layout : loyalty history moves"
+        inherit_id="portal.portal_breadcrumbs">
+        <xpath expr="//ol[hasclass('o_portal_submenu')]" position="inside">
+            <li class="breadcrumb-item" t-if="page_name == 'loyalty_history'">
+                <a t-attf-href="/my/history?{{ keep_query() }}">History</a>
+            </li>
+        </xpath>
+    </template>
+
+    <template id="portal_my_history" name="My Loyalty History">
+        <t t-call="portal.portal_layout">
+            <t t-set="breadcrumbs_searchbar" t-value="True"/>
+            <t t-call="portal.portal_searchbar">
+                <t t-set="title">Loyalty Transaction</t>
+            </t>
+            <t t-if="not coupon_history_lines">
+                <p class="alert alert-warning">
+                    There are currently no transaction lines for this Card.
+                </p>
+            </t>
+            <t t-else="" t-call="portal.portal_table">
+                <thead>
+                    <tr class="active">
+                      <th>Description</th>
+                      <th>Order</th>
+                      <th>Date</th>
+                      <th>Issued</th>
+                      <th>Used</th>
+                      <th>New Balance</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <t t-foreach="coupon_history_lines" t-as="move">
+                      <tr>
+                            <td><span t-out="move.description"></span></td>
+                            <td t-if="move.order_id">
+                                <a t-if="move.order_id._name == 'sale.order'" t-att-href="move.order_id.sudo().get_portal_url()">
```

Comment:
may be this giving missing record error :thinking: 
![image](https://github.com/user-attachments/assets/7e068d1d-467d-40ad-8414-0bb217d4bbc8)
![image](https://github.com/user-attachments/assets/7002da2e-957b-4017-bdee-2aae79cf7b88)
when history does not have order field set